src/Backend/Modules/Pages/Domain/Page/PageRepository.php

```diff
@@ -222,6 +222,71 @@ public function getFirstChild(int $pageId, string $status, string $language): ?P
             ->getOneOrNullResult();
     }
 
+    public function findOneByParentsAndUrlAndStatusAndLanguage(
+        array $parentIds,
+        string $url,
+        string $status,
+        string $language
+    ): ?Page {
+        $qb = $this
+            ->createQueryBuilder('p')
+            ->join('p.meta', 'meta');
+
+        $qb
+            ->where($qb->expr()->in('p.parentId', ':parentIds'))
+            ->andWhere('p.status = :status')
+            ->andWhere('meta.url = :url')
+            ->andWhere('p.language = :language');
+
+        $qb->setParameters(
+            [
+                'parentIds' => $parentIds,
+                'status' => $status,
+                'url' => $url,
+                'language' => $language,
+            ]
+        );
+
+        return $qb
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getOneOrNullResult();
+    }
+
+    public function findOneByParentsAndUrlAndStatusAndLanguageExcludingId(
+        array $parentIds,
+        string $url,
+        string $status,
+        string $language,
```

Comment:
I would use the Locale value object here

Reply:
See my previous comment regarding this.